### PR TITLE
add turn by turn navigator

### PR DESCRIPTION
**Pull request by SamuelRioTz**
implamentation of #386

https://github.com/user-attachments/assets/10ed1d1e-8901-4697-ba11-9ce25b64c027

### CODE REVIEW

**Comment by hbruch** [comment]
Thanks for posting the current state. Is the arrow taking into account device orientation (which it should) or just using the orientation of the underlying route suggestion at the position?

File: lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart [hunk 1]
```diff
@@ -55,6 +60,35 @@ class ItineraryDetailsCard extends StatelessWidget {
...
+                GestureDetector(
+                  child: Container(
+                    padding: EdgeInsets.symmetric(horizontal: 5, vertical: 2),
+                    decoration: BoxDecoration(
+                        borderRadius: BorderRadius.circular(20),
+                        border: Border.all()),
+                    child: Row(
+                      mainAxisSize: MainAxisSize.min,
+                      children: [
+                        Text("Start"),
```
Comment (SamuelRioTz) [inline lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart:72]: @GustavoTCh add translation pls

File: lib/base/pages/home/transport_selector/mode_tracker_screen.dart [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final mapRouteState = context.read<MapRouteCubit>().state;
+    final mapModesCubit = context.watch<MapModesCubit>();
+    final mapConfiguratiom = context.read<MapConfigurationCubit>().state;
+    return BaseTrufiPage(
+      child: Scaffold(
+        appBar: AppBar(title: Text("Turn by turn navigation")),
```
Comment (SamuelRioTz) [inline lib/base/pages/home/transport_selector/mode_tracker_screen.dart:368]: @GustavoTCh please add translation

**Comment by SamuelRioTz** [comment]
https://github.com/user-attachments/assets/f533520c-9482-4316-aec1-b078af64e769

**Comment by Denis306** [comment]
Hi @SamuelRioTz please adjust the coloring:
- The text in the "Start" Button should be green like in the button "Buy tickets"
image

- The text in the box with the route indications should be white.
image

**Comment by SamuelRioTz** [comment]
@Denis306 
image
image

File: lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart [hunk 3]
```diff
@@ -55,6 +60,45 @@ class ItineraryDetailsCard extends StatelessWidget {
...
+                    ),
+                    onTap: () async {
+                      final locationProvider = GPSLocationProvider();
+                      await locationProvider.startLocation(context);
+
+                      await Navigator.of(context).push(MaterialPageRoute(
+                        builder: (BuildContext context) => ModeTrackerScreen(
+                          title: localizationSB.navigationTurnByTurnNavigation,
+                          warning:
+                              'You have strayed too far from the route. Please return to the path.',
```
Comment (SamuelRioTz) [inline lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart:95]: @GustavoTCh  please translate it

**Comment by stadtnavimanager** [comment]
@SamuelRioTz in German you can use the word "Starten" instead of "Beginn"

File: lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart (hunk 3 above)
Comment (SamuelRioTz) [reply lib/base/pages/home/widgets/plan_itinerary_tabs/itinerary_details_card/itinerary_details_card.dart]: @GustavoTCh  https://github.com/stadtnavi/stadtnavi_app/pull/391#issuecomment-2556464814

**Review by GustavoTCh — APPROVED** [review]

**Pull request merged**